lib/jnpr/junos/facts/__init__.py

```diff
@@ -1,24 +1,108 @@
-from jnpr.junos.facts.chassis import facts_chassis
-from jnpr.junos.facts.routing_engines import facts_routing_engines
-from jnpr.junos.facts.personality import facts_personality
-from jnpr.junos.facts.swver import facts_software_version
-from jnpr.junos.facts.ifd_style import facts_ifd_style
-from jnpr.junos.facts.switch_style import facts_switch_style
-from jnpr.junos.facts.session import facts_session
-from jnpr.junos.facts.srx_cluster import facts_srx_cluster
-from jnpr.junos.facts.domain import facts_domain
-
-FACT_LIST = [
-    facts_chassis,                  # first
-    facts_routing_engines,          # second
-    facts_personality,              # third
-    facts_srx_cluster,              # four
-    facts_software_version,         # fifth
-
-    facts_domain,
-    facts_ifd_style,
-    facts_switch_style,
-    facts_session
-]
-
-__all__ = ['FACT_LIST']
+"""
+This static string is replaced by the dynamic __doc__ variable on import.
+"""
+import importlib
+import os
+import sys
+
+
+def _get_list_of_fact_module_names():
+    """
+    Get a list of fact module names.
+
+    Gets a list of the module names that reside in the facts directory (the
+    directory where this jnpr.junos.facts.__init__.py file lives). Any module
+    names that begin with an underscore (_) are ommitted.
+
+    :returns:
+      A list of fact module names.
+    """
+    module_names = []
+    facts_dir = os.path.dirname(__file__)
+    for file in os.listdir(facts_dir):
+        if (os.path.isfile(os.path.join(facts_dir, file)) and
+           not os.path.islink(os.path.join(facts_dir, file))):
+            if file.endswith('.py') and not file.startswith('_'):
```

Comment:
I did consider using `glob.glob()`, but I ultimately chose this method because there's no single glob pattern which would satisfy `file.endswith('.py') and not file.startswith('_')`. (At least I'm not aware of a way to specify negation in a glob pattern.) 

In addition, `glob.glob('*.py')` returns the .py files in the current directory rather than in `facts_dir`. 

If I do:
```
for file in glob.glob(facts_dir + '/*.py'):
```
Then the value of file:
1) contains the path which has to be stripped off in an extra step.
2) still has to be checked to make sure it's not a directory and not a link.
3) still has to be checked to make sure it doesn't start with `'_'`.